test/integration/tests/policyauthorize.sh

```diff
@@ -0,0 +1,102 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of Intel Corporation nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+#;**********************************************************************;
+
+source helpers.sh
+
+pcr_ids="0"
+alg_pcr_policy=sha256
+file_pcr_value=pcr.bin
+file_policy=policy.data
+file_authorized_policy_1=auth_policy_1.data
+file_authorized_policy_2=auth_policy_2.data
+file_session_file="session.dat"
+file_private_key="private.pem"
+file_public_key="public.pem"
+file_verifying_key_public="verifying_key_public"
+file_verifying_key_name="verifying_key_name"
+file_verifying_key_ctx="verifying_key_ctx"
+file_policyref="policyref"
+
+cleanup() {
+  rm -f  $file_pcr_value $file_policy $file_session_file $file_private_key \
+    $file_public_key $file_verifying_key_public $file_verifying_key_name \
+    $file_verifying_key_ctx $file_policyref $file_authorized_policy_1 \
+    $file_authorized_policy_2
+
+  tpm2_flushcontext -S $file_session_file 2>/dev/null || true
+}
+trap cleanup EXIT
+
+start_up
+
+cleanup
+
+generate_policyAuthorize () {
+
+  tpm2_startauthsession -Q -S $file_session_file
+
+  tpm2_policyauthorize -Q -S $file_session_file  -f $3 -a $1 -b $2 -c $4
+   
+  tpm2_flushcontext -S $file_session_file 
+}
+
+openssl genrsa -out $file_private_key 2048 2>/dev/null
+openssl rsa -in $file_private_key -out $file_public_key -pubout 2>/dev/null
+tpm2_loadexternal -G rsa -a n -u $file_public_key -o $file_verifying_key_ctx
+tpm2_readpublic -c $file_verifying_key_ctx | grep 'name:' |\
+grep -v q | awk '{print $2}' | xxd -r -p > $file_verifying_key_name
```

Comment:
use the yaml helpers in helpers.sh

Reply:
@williamcroberts thinking about this, actually the load external needs one another thing. It is actually capable of dumping the name .
https://github.com/tpm2-software/tpm2-tools/blob/90e5d99ed3771e1a390839d0ba38a6379757072e/tools/tpm2_loadexternal.c#L85
This will avoid calling read public altogether.

Reply:
Referencing issue #1163